pkg/cloudprovider/providers/openstack/openstack_routes.go

```diff
@@ -276,3 +277,41 @@ func (r *Routes) DeleteRoute(clusterName string, route *cloudprovider.Route) err
 	onFailure.Disarm()
 	return nil
 }
+
+func getPortByID(client *gophercloud.ServiceClient, portID string) (neutronports.Port, error) {
+	var port neutronports.Port
+	targetPort, err := neutronports.Get(client, portID).Extract()
+	if err != nil {
+		return port, err
+	}
+
+	if targetPort == nil {
+		return port, ErrNotFound
+	}
+
+	port = *targetPort
```

Comment:
This function will be called by CreateRoute and DeleteRoute, we can collect these code into a func().